### PR TITLE
remove config extract magic

### PR DESCRIPTION
**Pull request by Segaja**
relates to https://github.com/vcs-python/vcspull/issues/360

- tests not working yet
- auto completion is most likely broken
- removed `vcspull sync <url>`
- documentation for config change missing

### CODE REVIEW

**Comment by Segaja** [comment]
```yaml
${HOME}/develop/github.com/vcs-python/checkouts:
  dotfiles:
    vcs: git
    options:
      remotes:
        origin: ssh://git@gitlab.com/Segaja/dotfiles.git
  procpath:
    vcs: hg
    options:
      url: https://heptapod.host/saajns/procpath
  subversion:
    vcs: svn
    options:
      url: https://svn.apache.org/repos/asf/subversion/trunk
~/develop/github.com/vcs-python/checkouts2:
  discordbot:
    vcs: git
    options:
      remotes:
        origin: ssh://git@gitlab.com/Segaja/discordbot.git
```
```yaml
~/develop/github.com/vcs-python/checkouts3:
  discordbot:
    vcs: git
    options:
      remotes:
        origin: ssh://git@gitlab.com/Segaja/discordbot.git
```

**Comment by tony** [comment]
@Segaja Nice rebase. 💪

File: vcspull/cli/sync.py [hunk 1]
```diff
@@ -67,40 +68,37 @@ def sync(repo_terms, config):
...
 
-    list(map(update_repo, found_repos))
+    for path, repos in found_repos.items():
+        for name, repo in repos.items():
+            r: BaseProject = DEFAULT_VCS_CLASS_MAP[repo["vcs"]](
+                repo_dir=os.path.join(path, name),
+                options=repo["options"],
+                progress_callback=progress_cb,
+            )
+            r.update_repo(set_remotes=True)
```
Comment (tony) [inline vcspull/cli/sync.py:99]: @Segaja I'm using a derivative of this in #373

File: vcspull/config.py [hunk 2]
```diff
@@ -195,88 +127,96 @@ def find_config_files(
...
-    path_duplicates = list(set(curpaths).intersection(newpaths))
+    for parent_path, repos in config2.items():
+        if parent_path in config1:
+            for name, repo in repos.items():
+                if name in config1[parent_path]:
+                    dupes += (repo, config1[parent_path][name])
 
-    if not path_duplicates:
-        return None
+    return dupes
```
Comment (tony) [inline vcspull/config.py:195]: This is the area I used in #373 (but changed a lot)

**Pull request closed without merging**